lib/util/util.HashMessageProvider.js

```diff
@@ -0,0 +1,45 @@
+this.util = this.util || {};
+
+util.HashMessageProvider = ( function() {
+	'use strict';
+
+	/**
+	 * Providing messages using specified default message
+	 * @class util.HashMessageProvider
+	 * @licence GNU GPL v2+
+	 * @author H. Snater < mediawiki@snater.com >
+	 * @author Adrian Heine < adrian.heine@wikimedia.de >
+	 *
+	 * @constructor
+	 *
+	 * @param {Object} messages
+	 */
+	function HashMessageProvider( messages ) {
+		this._messages = messages;
+	}
+
+	HashMessageProvider.prototype = {
+		constructor: HashMessageProvider,
+
+		/**
+		 * @property {Object} [_messages={}]
+		 * @private
+		 */
+		_messages: null,
+
+		/**
+		 * Returns a message from the pre-defined hash
+		 *
+		 * @param {string} key
+		 * @param {string[]} [params=[]] Message parameters.
+		 * @return {string|null}
+		 */
+		getMessage: function( key, params ) {
```

Comment:
This ignores `params`. If intended (I think it is) then please describe this in the documentation.